### PR TITLE
allows users to remove config options with dc.configure

### PR DESCRIPTION
**Pull request by karissa**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -105,63 +105,22 @@ class DeltaChat extends EventEmitter {
...
-    }
+    this.setConfig('send_server', opts.send_server)
+    this.setConfig('send_user', opts.send_user)
+    this.setConfig('send_pw', opts.send_pw)
+    this.setConfig('send_port', String(opts.send_port))
+    this.setConfig('server_flags', String(opts.server_flags))
+    this.setConfig('displayname', opts.displayname)
+    this.setConfig('selfstatus', opts.selfstatus)
+    this.setConfig('selfavatar', opts.selfavatar)
+    this.setConfig('e2ee_enabled', String(opts.e2ee_enabled ? 1 : 0))
```
Comment (ralphtheninja) [inline index.js]: This means that `e2ee_enabled` will default to `false` which might be a problem, since in core it defaults to true.

Comment (ralphtheninja) [reply index.js]: Also, the c side cries when passing `undefined`. We should probably add a guard in `setConfig()`, like:

```js
setConfig (key, value) {                                     
  return binding.dcn_set_config(this.dcn_context, key || '', value || '')
}                                                            
```

**Comment by ralphtheninja** [comment]
What is the problem we're solving here?

File: index.js (hunk 1 above)
Comment (karissa) [reply index.js]: @ralphtheninja but @r10s says that NULL should be accepted by core.

This is to implement the `advanced` settings in the UI. If you set some option, and then want to remove it later, you have no way of doing that now.

File: index.js [hunk 2]
```diff
@@ -103,65 +104,26 @@ class DeltaChat extends EventEmitter {
 
     this.once('_configured', ready)
 
-    this.setConfig('addr', opts.addr)
-
-    if (typeof opts.mail_server === 'string') {
-      this.setConfig('mail_server', opts.mail_server)
-    }
+    if (!opts.e2ee_enabled) opts.e2ee_enabled = 1
```
Comment (ralphtheninja) [inline index.js]: Doing `opts.e2ee_enabled = false` will make it true. Maybe check for `undefined` instead?

File: index.js (hunk 1 above)
Comment (ralphtheninja) [reply index.js]: > @ralphtheninja but @r10s says that NULL should be accepted by core.

Aye. So we already do this for other functions, but pass empty string and check for it on the c side, e.g. https://github.com/deltachat/deltachat-node/blob/master/src/module.c#L695

**Pull request merged**